ocs_ci/ocs/registry.py

```diff
@@ -0,0 +1,282 @@
+            time.sleep(wait_time)
+    return pod_obj
+
+
+def get_oc_podman_login_cmd():
+    """
+    Function to get oc and podman login commands on node
+
+    Returns:
+        cmd_list (list): List of cmd for oc/podman login
+    """
+    user = config.RUN['username']
+    filename = os.path.join(
+        config.ENV_DATA['cluster_path'],
+        config.RUN['password_location']
+    )
+    with open(filename) as f:
+        password = f.read()
+    helpers.refresh_oc_login_connection()
+    ocp_obj = ocp.OCP()
+    token = ocp_obj.get_user_token()
+    route = get_default_route_name()
+    cmd_list = [
+        'export KUBECONFIG=/home/core/auth/kubeconfig',
+        f"oc login -u {user} -p {password}",
+        f"podman login {route} -u {user} -p {token}"
+    ]
+    master_list = helpers.get_master_nodes()
+    helpers.rsync_kubeconf_to_node(node=master_list[0])
+    return cmd_list
+
+
+def validate_pvc_mount_on_registry_pod():
+    """
+    Function to validate pvc mounted on the registry pod
+
+    Raises:
+        AssertionError: When PVC mount not present in the registry pod
+    """
+    pod_obj = get_registry_pod_obj()
+    mount_point = pod_obj[0].exec_cmd_on_pod(command="mount")
+    assert "/registry" in mount_point, f"pvc is not mounted on pod {pod_obj.name}"
+    logger.info("Verified pvc is mounted on image-registry pod")
+
+
+def validate_registry_pod_status():
+    """
+    Function to validate registry pod status
+    """
+    pod_obj = get_registry_pod_obj()
+    helpers.wait_for_resource_state(pod_obj[0], state=constants.STATUS_RUNNING)
+
+
+def validate_all_image_registry_pod_status():
+    """
+    Function to validate all image-registry pod status
+    """
+    pass
+
+
+def validate_data_at_backend():
+    """
+    Function to validate image_push data from ceph
+    """
+    pass
+
+
+def get_registry_pvc():
+    """
+    Function to get registry pvc
+
+    Returns:
+        pvc_name (str): Returns name of the OCS pvc backed for registry
+    """
+    pod_obj = get_registry_pod_obj()
+    return pod.get_pvc_name(pod_obj)
+
+
+def get_default_route_name():
+    """
+    Function to get default route name
+
+    Returns:
+        route_name (str): Returns default route name
+    """
+    ocp_obj = ocp.OCP()
+    route_cmd = f"get route -n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE} -o yaml"
+    route_dict = ocp_obj.exec_oc_cmd(command=route_cmd)
+    return route_dict.get('items')[0].get('spec').get('host')
+
+
+def add_role_to_user(role_type, user):
+    """
+    Function to add role to user
+
+    Args:
+        role_type (str): Type of the role to be added
+        user (str): User to be added for the role
+
+    Raises:
+        AssertionError: When failure in adding new role to user
+    """
+    ocp_obj = ocp.OCP()
+    role_cmd = f"policy add-role-to-user {role_type} {user} -n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE}"
+    assert ocp_obj.exec_oc_cmd(command=role_cmd), f"Adding role failed"
```

Comment:
I see you are trying format string but no variable formatted. f"Adding role failed" . So remove f if no need to format here